src/languages/yaml.js

```diff
@@ -107,20 +107,34 @@ export default function(hljs) {
     relevance: 0
   };
 
+  // Optimized block scalar handling to avoid polynomial backtracking
+  const BLOCK_SCALAR = {
+    className: 'string',
+    begin: /[|>][-+\d]*\s*$/, // Block scalar indicator with modifiers
```

Comment:
Is there a reason the regex here around +-number is much fuzzier than the tighter one below?

Reply:
are you referring to the regex on line 113 and 116 ?

Reply:
Referring to this looking much more correct:

`'[\\|>]([-+]?[0-9]*)?[ ]*\\n'`

ie, +- followed by digits, etc...

This regest would allow all sorts of `-43+23423---13+++`...